### ISSUE TITLE
Documentation of ctc_beam_search_decoder is inconsistent

### ISSUE THREAD

**Issue by ahhentz** [tensorflow/tensorflow#6929]
The documentation of ctc_beam_search_decoder describes the `input` argument as:

    inputs: 3-D `float` `Tensor`, size
        `[max_time x batch_size x num_classes]`.  The logits.

Implying that they should be the linear projections. However, by inspecting the [unit tests](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/kernel_tests/ctc_decoder_ops_test.py#L96) and the [implementation](https://github.com/tensorflow/tensorflow/blob/679f95e9d8d538c3c02c0da45606bab22a71420e/tensorflow/core/util/ctc/ctc_beam_search.h#L290), it seems these inputs should actually be log-probabilities. In this case, the documentation should read:

    inputs: 3-D `float` `Tensor`, size
      `[max_time x batch_size x num_classes]`.
    The output symbol log-probabilities.

**Comment by asimshankar** [comment]
@ebrevdo : Would you like to add some clarifications?

**Comment by gunan** [comment]
@MarkDaoust Could you take a look?

**Comment by MarkDaoust** [comment]
I don't see a problem here.

Feel free to correct me if I'm wrong, but "logits" seems to be used throughout TensorFlow to refer to  un-normalized log probabilities like these. see [softmax_cross_entropy_with_logits] (https://www.tensorflow.org/api_docs/python/tf/nn/softmax_cross_entropy_with_logits) and friends for example.

In the test, [line 191](https://github.com/tensorflow/tensorflow/blob/r1.2/tensorflow/python/kernel_tests/ctc_decoder_ops_test.py#L191), we see that this exactly what they're passing in. The rows start out summing to 1, then they take the log and add an arbitrary offset.

I'm not familiar with the implementation details, but the first thing they do in the implementation is subtract the max (for numerical stability).
